src/jormungandr/jormungandr/src/network/p2p/comm.rs

```diff
@@ -465,23 +466,44 @@ pub struct PeerInfo {
     pub stats: PeerStats,
 }
 
+/// Future for executing async address lookup
+/// for the given node id from [Peers].
+pub struct GetNodeAddress {
```

Comment:
what happens when this can't resolve?

Reply:
That's right. I just kept the same behavior as before: https://github.com/input-output-hk/catalyst-core/pull/241/files/d284ff271c9c130b5833e64a025b4528125566eb#diff-c9c45cadddc07944c423cf18d0c0775c7fda7d1c69e0ec690e21fa8bcda5e0b8L316-L330